cdap-api-common/src/main/java/co/cask/cdap/api/data/format/StructuredRecord.java

```diff
@@ -113,32 +277,138 @@ public Builder set(String fieldName, @Nullable Object value) {
       return this;
     }
 
+    /**
+     * Sets the date value for {@link Schema.LogicalType#DATE} field
+     *
+     * @param fieldName name of the field to set
+     * @param localDate value for the field
+     * @return this builder
+     * @throws UnexpectedFormatException if the field is not in the schema, or the field is not nullable but a null
+     *                                   value is given
+     * @throws IllegalArgumentException if the provided date is an invalid date
+     */
+    public Builder setDate(String fieldName, @Nullable LocalDate localDate) {
+      Schema.Field field = validateAndGetField(fieldName, localDate);
+      validateAndGetLogicalType(fieldName, field,
+                                new HashSet<>(Collections.singletonList(Schema.LogicalType.DATE.name())));
+      if (localDate == null) {
+        fields.put(fieldName, null);
```

Comment:
Why need to put a null? Isn't that a missing key in the fields means the value is `null`?

Reply:
If we do not add this key check in `build()` https://github.com/caskdata/cdap/blob/develop/cdap-api-common/src/main/java/co/cask/cdap/api/data/format/StructuredRecord.java#L189 would throw exception.